includes/api/class-give-api.php

```diff
@@ -1735,6 +1735,10 @@ public function output( $status_code = 200 ) {
 	function user_key_field( $user ) {
 
 		if ( ( give_get_option( 'api_allow_user_keys', false ) || current_user_can( 'manage_give_settings' ) ) && current_user_can( 'edit_user', $user->ID ) ) {
+
+			// Flush Rewrite Rules to avoid 404 error using Give API for first time.
+			flush_rewrite_rules();
```

Comment:
@mehul0810 this isn't the proper place to hook in because it would call `flush_rewrite_rules()` way too many times and that's resource exhaustive and not best practice. I suggest only hooking in a _new_ API is created. Most likely within the `generate_api_key()` method.